library/entity/entity/src/main/java/org/quiltmc/qsl/entity/api/QuiltEntityTypeBuilder.java

```diff
@@ -0,0 +1,477 @@
+ * server-&gt;client entity tracking values.
+ *
+ * @param <T> Entity class
+ */
+public class QuiltEntityTypeBuilder<T extends Entity> {
+	private SpawnGroup spawnGroup;
+	private EntityType.EntityFactory<T> factory;
+	private boolean saveable = true;
+	private boolean summonable = true;
+	private int trackRange = 5;
+	private int trackedUpdateRate = 3;
+	private Boolean forceTrackedVelocityUpdates = null;
+	private boolean fireImmune = false;
+	private boolean spawnableFarFromPlayer;
+	private EntityDimensions dimensions = EntityDimensions.changing(-1.0f, -1.0f);
+	private ImmutableSet<Block> specificSpawnBlocks = ImmutableSet.of();
+
+	protected QuiltEntityTypeBuilder(SpawnGroup spawnGroup, EntityType.EntityFactory<T> factory) {
+		this.spawnGroup = spawnGroup;
+		this.factory = factory;
+		this.spawnableFarFromPlayer = spawnGroup == SpawnGroup.CREATURE || spawnGroup == SpawnGroup.MISC;
+	}
+
+	/**
+	 * Creates an entity type builder.
+	 *
+	 * <p>This entity's spawn group will automatically be set to {@link SpawnGroup#MISC}.</p>
+	 *
+	 * @param <T> the type of entity
+	 *
+	 * @return a new entity type builder
+	 */
+	public static <T extends Entity> QuiltEntityTypeBuilder<T> create() {
+		return create(SpawnGroup.MISC);
+	}
+
+	/**
+	 * Creates an entity type builder.
+	 *
+	 * @param spawnGroup the entity spawn group
+	 * @param <T> the type of entity
+	 *
+	 * @return a new entity type builder
+	 */
+	public static <T extends Entity> QuiltEntityTypeBuilder<T> create(SpawnGroup spawnGroup) {
+		return create(spawnGroup, QuiltEntityTypeBuilder::emptyFactory);
+	}
+
+	/**
+	 * Creates an entity type builder.
+	 *
+	 * @param spawnGroup the entity spawn group
+	 * @param factory the entity factory used to create this entity
+	 * @param <T> the type of entity
+	 *
+	 * @return a new entity type builder
+	 */
+	public static <T extends Entity> QuiltEntityTypeBuilder<T> create(SpawnGroup spawnGroup, EntityType.EntityFactory<T> factory) {
+		return new QuiltEntityTypeBuilder<>(spawnGroup, factory);
+	}
+
+	/**
+	 * Creates an entity type builder for a living entity.
+	 *
+	 * <p>This entity's spawn group will automatically be set to {@link SpawnGroup#MISC}.</p>
+	 *
+	 * @param <T> the type of entity
+	 *
+	 * @return a new living entity type builder
+	 */
+	public static <T extends LivingEntity> QuiltEntityTypeBuilder.Living<T> createLiving() {
+		return new QuiltEntityTypeBuilder.Living<>(SpawnGroup.MISC, QuiltEntityTypeBuilder::emptyFactory);
+	}
+
+	/**
+	 * Creates an entity type builder for a mob entity.
+	 *
+	 * @param <T> the type of entity
+	 *
+	 * @return a new mob entity type builder
+	 */
+	public static <T extends MobEntity> QuiltEntityTypeBuilder.Mob<T> createMob() {
+		return new QuiltEntityTypeBuilder.Mob<>(SpawnGroup.MISC, QuiltEntityTypeBuilder::emptyFactory);
+	}
+
+	private static <T extends Entity> T emptyFactory(EntityType<T> type, World world) {
```

Comment:
There are missing Javadocs on many of these methods; These should have something